azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_azurerm.go

```diff
@@ -263,8 +303,32 @@ func resourceServiceEndpointAzureRMUpdate(d *schema.ResourceData, m interface{})
 			return err
 		}
 	}
-	_, err = updateServiceEndpoint(clients, serviceEndpoint)
-
+    // Handle shared_project_ids updates
+    if d.HasChange("shared_project_ids") {
+        old, new := d.GetChange("shared_project_ids")
+        oldSet := toStringSet(old.([]interface{}))
```

Comment:
The built-in function `schema.NewSet(f SchemaSetFunc, items []interface{}) *Set` can be used to process Set collections.